### PR TITLE
[DXFC-376]  Add an example with the ability to specify from-time and to-time arguments

### PR DESCRIPTION
**Pull request by stdcion**

### CODE REVIEW

File: samples/SnapshotConsoleSample/SnapshotConsoleSample.c [hunk 1]
```diff
@@ -474,7 +569,7 @@ int main(int argc, char *argv[]) {
 			"of events and a full order book.\n"
 			"-------------------------------------------------------------------------------\n"
 			"Usage: SnapshotConsoleSample <server address> <event type> <symbol>\n"
-			"       [order_source] [" TIME_PARAM_SHORT_TAG " <DD-MM-YYYY>] "
+			"       [order_source] [" FROM_TIME_PARAM_SHORT_TAG " <DD-MM-YYYY>] [" TO_TIME_PARAM_SHORT_TAG " <DD-MM-YYYY>]"
```
Comment (ttldtor) [inline samples/SnapshotConsoleSample/SnapshotConsoleSample.c:572]: This time format does not correspond to the format specified below.

File: samples/SnapshotConsoleSample/SnapshotConsoleSample.c [hunk 2]
```diff
@@ -485,15 +580,16 @@ int main(int argc, char *argv[]) {
...
+			"  " FROM_TIME_PARAM_SHORT_TAG " <DD-MM-YYYYTHH:mm:ss>  - Time from which to receive data (default: current date and time\n"			
+			"  " TO_TIME_PARAM_SHORT_TAG "   <DD-MM-YYYYTHH:mm:ss>  - Time to which to receive data (default: infinity, exist only for CANDLE, TIME_AND_SALE and GREEKS)\n"
 			"  " RECORDS_PRINT_LIMIT_SHORT_PARAM " <limit>       - The number of displayed records "
 			"(0 - unlimited, default: " STRINGIFY(DEFAULT_RECORDS_PRINT_LIMIT)")\n"
 			"  " TOKEN_PARAM_SHORT_TAG " <token>       - The authorization token\n"
 			"  " LOG_DATA_TRANSFER_TAG "               - Enables the data transfer logging\n"
 			"  " TIMEOUT_TAG " <timeout>     - Sets the program timeout in seconds (default = 604800,\n"
 			"                     i.e a week)\n"
-			"Example: demo.dxfeed.com:7300 ORDER IBM NTV -t 01-01-1970 -o 30\n\n"
+			"Example: demo.dxfeed.com:7300 ORDER IBM NTV -from-time 01-01-1970T00:00:00 -o 30\n\n"
```
Comment (ttldtor) [inline samples/SnapshotConsoleSample/SnapshotConsoleSample.c:591]: We should specify that user must don't use `-from-time` `-to-time` with "indexed only" (indexed but not time series) events (Order, Spread Order, Series).

File: samples/SnapshotConsoleSample/SnapshotConsoleSample.c [hunk 3]
```diff
@@ -671,7 +784,8 @@ int main(int argc, char *argv[]) {
 			return 32;
 		}
 	} else {
-		if (!dxf_create_snapshot(connection, event_id, base_symbol, NULL, time_is_set ? time_value * 1000 : 0, &snapshot)) {
+		if (!dxf_create_snapshot(connection, event_id, base_symbol, NULL, from_time_is_set ? from_time_value * 1000 : 0,
```
Comment (ttldtor) [inline samples/SnapshotConsoleSample/SnapshotConsoleSample.c:787]: And probably we should add a check for non "indexed only" events here.
`Order, Spread Order, Series: from_time = 0`

File: samples/SnapshotConsoleSample/SnapshotConsoleSample.c [hunk 4]
```diff
@@ -404,45 +515,28 @@ int atoi2(char *str, int *result) {
 	return true;
 }
 
-#define DATE_TIME_BUF_SIZE 4
 /*
  * Parse date string in format 'DD-MM-YYYY'
```
Comment (ttldtor) [inline samples/SnapshotConsoleSample/SnapshotConsoleSample.c:519]: DD-MM-YYYYTHH:mm:ss

**Review by ttldtor — CHANGES_REQUESTED** [review]

File: samples/SnapshotConsoleSample/SnapshotConsoleSample.c [hunk 5]
```diff
@@ -259,124 +272,222 @@ dxf_string_t ansi_to_unicode(const char *ansi_str) {
...
+			L"option volume=%.15g, put call ratio=%.15g, forward_price=%.15g, dividend=%.15g, interest=%.15g, "
+			L"index=0x%" LS(PRIX64) L", flags=0x%X}\n",
+			srs.sequence, srs.expiration, srs.volatility, srs.call_volume, srs.put_volume, srs.option_volume,
+			srs.put_call_ratio, srs.forward_price, srs.dividend, srs.interest, srs.index, srs.event_flags);
+	}
+}
+
+void listener(const dxf_snapshot_data_ptr_t snapshot_data, void *user_data) {
+	size_t i;
+	size_t records_count = snapshot_data->records_count;
```
[the remark this answers is not in the document]
Comment (stdcion) [reply samples/SnapshotConsoleSample/SnapshotConsoleSample.c]: Done

File: samples/SnapshotConsoleSample/SnapshotConsoleSample.c (hunk 1 above)
Comment (stdcion) [reply samples/SnapshotConsoleSample/SnapshotConsoleSample.c]: Done

File: samples/SnapshotConsoleSample/SnapshotConsoleSample.c (hunk 2 above)
Comment (stdcion) [reply samples/SnapshotConsoleSample/SnapshotConsoleSample.c]: Done

File: samples/SnapshotConsoleSample/SnapshotConsoleSample.c (hunk 3 above)
Comment (stdcion) [reply samples/SnapshotConsoleSample/SnapshotConsoleSample.c:791]: Done

File: samples/SnapshotConsoleSample/SnapshotConsoleSample.c (hunk 4 above)
Comment (stdcion) [reply samples/SnapshotConsoleSample/SnapshotConsoleSample.c]: Done

**Review by ttldtor — APPROVED** [review]

**Pull request merged**